### ISSUE TITLE
Erratic results when training NER using transformers on nightly

### ISSUE THREAD

**Issue by fcggamou** [explosion/spaCy#6233]
Hi,

I'm experimenting with training a NER model using the nightly version. 
Training using a tok2vec component worked smooth as butter, and I got similar results as with Spacy 2.3.2:

```
✔ Created output directory: model
ℹ Using GPU: 0

=========================== Initializing pipeline ===========================
Set up nlp object from config
Pipeline: ['tok2vec', 'ner']
Created vocabulary
Finished initializing nlp object
Initialized pipeline components: ['tok2vec', 'ner']
✔ Initialized pipeline

============================= Training pipeline =============================
ℹ Pipeline: ['tok2vec', 'ner']
ℹ Initial learn rate: 0.001
E     #        LOSS TOK2VEC   LOSS NER   ENTS_F   ENTS_P   ENTS_R   SCORE 
---   ------   ------------   --------   ------   ------   ------   ------
  0        0           0.00      41.00     3.71     5.72     2.75     0.04
  0      200         120.55    5192.01    56.22    59.73    53.10     0.56
  0      400         138.67    4601.48    59.48    65.31    54.60     0.59
  0      600         177.05    5215.84    64.26    63.38    65.17     0.64
  0      800         204.31    6169.24    65.38    68.48    62.55     0.65
  0     1000         250.86    7031.01    68.23    67.04    69.46     0.68
  0     1200         323.35    8976.15    69.77    69.72    69.81     0.70
  1     1400         458.85    9596.12    69.81    69.60    70.01     0.70
  1     1600         551.70   11830.54    69.37    69.37    69.37     0.69
  2     1800         668.40   13477.82    71.58    69.54    73.73     0.72
  2     2000         901.45   15805.82    70.52    69.88    71.17     0.71
  3     2200        1247.45   17643.22    71.53    71.23    71.83     0.72
  4     2400        1546.50   19850.30    71.15    70.43    71.89     0.71
  5     2600        1799.25   19110.36    71.51    71.18    71.85     0.72
  6     2800        1950.51   18186.91    71.02    71.10    70.94     0.71
  7     3000        2111.22   17332.46    71.76    71.39    72.13     0.72
  8     3200        2315.25   16694.02    71.32    70.55    72.11     0.71
  8     3400        2360.38   15777.14    71.16    70.26    72.09     0.71
  9     3600        2459.01   15064.81    70.48    70.68    70.27     0.70
 10     3800        2543.32   14598.77    71.14    70.20    72.09     0.71
 11     4000        2645.07   13990.41    70.89    70.98    70.79     0.71
 12     4200        2689.16   13434.37    70.52    70.55    70.50     0.71
 13     4400        2788.04   13266.07    71.66    70.28    73.10     0.72
 14     4600        2968.43   12544.58    70.71    70.25    71.17     0.71
✔ Saved pipeline to output directory
```

However, using the same data to train the NER with the transformer component instead of tok2vec, with the default configuration, I'm getting some erratic results and huge transformer losses:

```
=========================== Initializing pipeline ===========================
Set up nlp object from config
Pipeline: ['transformer', 'ner']
Created vocabulary
Finished initializing nlp object
Downloading: 100% 456/456 [00:00<00:00, 337kB/s]
Downloading: 100% 248k/248k [00:00<00:00, 755kB/s]
Downloading: 100% 2.00/2.00 [00:00<00:00, 1.52kB/s]
Downloading: 100% 112/112 [00:00<00:00, 85.3kB/s]
Downloading: 100% 42.0/42.0 [00:00<00:00, 32.3kB/s]
Downloading: 100% 442M/442M [00:07<00:00, 61.4MB/s]
Initialized pipeline components: ['transformer', 'ner']
✔ Initialized pipeline

============================= Training pipeline =============================
ℹ Pipeline: ['transformer', 'ner']
ℹ Initial learn rate: 0.0
E     #        LOSS TRANSFORMER   LOSS NER   ENTS_F   ENTS_P   ENTS_R   SCORE 
---   ------   ----------------   --------   ------   ------   ------   ------
  0        0       450563616.00     177.58     8.21     9.43     7.28     0.08
  1      200   1218929133217557.00   242077.07     1.23     4.79     0.71     0.01
  2      400   949766226272769525785034752.00   267214.24    19.26    12.01    48.50     0.19
  3      600   919488244271205773737984.00   267968.07    17.80    11.46    39.86     0.18
  4      800    269577355312.37   245442.82     5.56     7.15     4.55     0.06
  6     1000   66217027905699856384.00   280587.19     0.54     1.96     0.32     0.01
  7     1200   3912179790620193193984.00   327658.82     6.34     9.60     4.73     0.06
  8     1400   100937580797255155712.00   243663.81    16.42    11.30    30.02     0.16
  9     1600   14769849285175922688.00   256728.50    12.64     9.36    19.46     0.13
 11     1800   9330143687895257448448.00   265807.64    11.60    11.27    11.94     0.12
 12     2000    252707409934.98   239694.04    15.95    10.95    29.35     0.16
✔ Saved pipeline to output directory
model_t/model-last
```

As you can see the accuracy goes up and down erratically and the final result is also pretty bad compared to the tok2vec based model.

Is this maybe a matter of me needing to fine-tune the hyperparameters in the config file? Or maybe is there any extra pre-processing requirement for transformer-like models?
This is the config that I'm using (it's the default generated by the init command):

```
[paths]
train = null
dev = null
vectors = null
init_tok2vec = null

[system]
gpu_allocator = "pytorch"
seed = 0

[nlp]
lang = "es"
pipeline = ["transformer","ner"]
tokenizer = {"@tokenizers":"spacy.Tokenizer.v1"}
disabled = []
before_creation = null
after_creation = null
after_pipeline_creation = null

[components]

[components.ner]
factory = "ner"
moves = null
update_with_oracle_cut_size = 100

[components.ner.model]
@architectures = "spacy.TransitionBasedParser.v1"
state_type = "ner"
extra_state_tokens = false
hidden_width = 64
maxout_pieces = 2
use_upper = false
nO = null

[components.ner.model.tok2vec]
@architectures = "spacy-transformers.TransformerListener.v1"
grad_factor = 1.0
pooling = {"@layers":"reduce_mean.v1"}

[components.transformer]
factory = "transformer"
max_batch_items = 4096
set_extra_annotations = {"@annotation_setters":"spacy-transformers.null_annotation_setter.v1"}

[components.transformer.model]
@architectures = "spacy-transformers.TransformerModel.v1"
name = "mrm8488/RuPERTa-base"

[components.transformer.model.get_spans]
@span_getters = "spacy-transformers.strided_spans.v1"
window = 128
stride = 96

[components.transformer.model.tokenizer_config]
use_fast = true

[corpora]

[corpora.dev]
@readers = "spacy.Corpus.v1"
path = ${paths.dev}
max_length = 0
gold_preproc = false
limit = 0
augmenter = null

[corpora.train]
@readers = "spacy.Corpus.v1"
path = ${paths.train}
max_length = 500
gold_preproc = false
limit = 0
augmenter = null

[training]
accumulate_gradient = 3
dev_corpus = "corpora.dev"
train_corpus = "corpora.train"
seed = ${system.seed}
gpu_allocator = ${system.gpu_allocator}
dropout = 0.1
patience = 1600
max_epochs = 0
max_steps = 20000
eval_frequency = 200
frozen_components = []
before_to_disk = null

[training.batcher]
@batchers = "spacy.batch_by_padded.v1"
discard_oversize = true
size = 2000
buffer = 256
get_length = null

[training.logger]
@loggers = "spacy.ConsoleLogger.v1"
progress_bar = false

[training.optimizer]
@optimizers = "Adam.v1"
beta1 = 0.9
beta2 = 0.999
L2_is_weight_decay = true
L2 = 0.01
grad_clip = 1.0
use_averages = false
eps = 0.00000001

[training.optimizer.learn_rate]
@schedules = "warmup_linear.v1"
warmup_steps = 250
total_steps = 20000
initial_rate = 0.00005

[training.score_weights]
ents_per_type = null
ents_f = 1.0
ents_p = 0.0
ents_r = 0.0

[pretraining]

[initialize]
vectors = null
init_tok2vec = ${paths.init_tok2vec}
vocab_data = null
lookups = null

[initialize.components]

[initialize.tokenizer]
```

Appreciate any feedback.

## Your Environment
- **spaCy version:** 3.0.0a36
- **Platform:** Linux-4.19.112+-x86_64-with-Ubuntu-18.04-bionic
- **Python version:** 3.6.9

**Comment by svlandeg** [comment]
This is currently not working properly yet - we've been looking into it. We plan on making an announcement when we get to a more stable nightly version ;-)

**Comment by fcggamou** [comment]
Thanks Sofie for the clarification. I will be eagerly waiting for the announcement in the meantime :)
